metadata-ingestion/src/datahub/ingestion/source/glue.py

```diff
@@ -0,0 +1,211 @@
+            )
+        elif self.aws_access_key_id and self.aws_secret_access_key:
+            return boto3.client(
+                'glue',
+                aws_access_key_id=self.aws_access_key_id,
+                aws_secret_access_key=self.aws_secret_access_key,
+                region_name=self.aws_region
+            )
+        else:
+            return boto3.client(
+                'glue',
+                region_name=self.aws_region
+            )
+
+
+@dataclass
+class GlueSourceReport(SourceReport):
+    tables_scanned = 0
+
+    def report_table_scanned(self) -> None:
+        self.tables_scanned += 1
+
+
+@dataclass
+class GlueSource(Source):
+    source_config: GlueSourceConfig
+    report = GlueSourceReport()
+
+    def __init__(self, config: GlueSourceConfig, ctx: PipelineContext):
+        super().__init__(ctx)
+        self.source_config = config
+        self.report = GlueSourceReport()
+        self.glue = config.glue
+        self.env = config.env
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = GlueSourceConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        logger.debug("listing tables")
+        tables = []
+        kwargs = {}
+        while True:
+            data = self.glue.search_tables(**kwargs)
+            tables += data['TableList']
+            if 'NextToken' in data:
+                kwargs['NextToken'] = data['NextToken']
+            else:
+                break
+        for table in tables:
+            table_name = table['Name']
+            database_name = table['DatabaseName']
+            full_table_name = f"{database_name}.{table_name}"
+
+            self.report.report_table_scanned()
+            mce = self._extract_record(table, full_table_name)
+            wu = MetadataWorkUnit(id=f"glue-{full_table_name}", mce=mce)
+            self.report.report_workunit(wu)
+            yield wu
+
+    def _extract_record(self, table, table_name) -> MetadataChangeEvent:
+        sys_time = int(time.time() * 1000)
+        metadata_record = MetadataChangeEvent()
+        dataset_snapshot = DatasetSnapshot(
+            urn=f"urn:li:dataset:(urn:li:dataPlatform:glue,{table_name},{self.env})",
+            aspects=[],
+        )
+
+        owner = table.get('Owner')
+        if owner:
+            owners = [
+                OwnerClass(
+                    owner=f"urn:li:corpuser:{owner}",
+                    type=OwnershipTypeClass.DATAOWNER
+                )
+            ]
+        else:
+            owners = []
+        dataset_snapshot.aspects.append(OwnershipClass(owners=owners, lastModified=AuditStampClass(
+            time=int(time.time() * 1000),
+            actor="urn:li:corpuser:datahub",
+        )))
+
+        dataset_snapshot.aspects.append(DatasetPropertiesClass(
+            description=table.get('Description'),
+            customProperties=table.get('Parameters', {}),
```

Comment:
I think there is more useful information in the StorageDescriptor and in the regular data which you are not using. You could provide this here as well as extra properties.

Reply:
So maybe change this to 
```customProperties={**table.get('Parameters', {}), **{ k: str(v) for k, v in table['StorageDescriptor'].items() if k != "columns"}```

Reply:
Maybe need to exclude a few more fields to avoid polluting it too much.

Reply:
oh that's a good idea. I might leave it as everything other than columns for the moment so all the info is available and see what people do / don't want to use, I imagine my use case is different to others